azext_iot/central/commands_destination.py

```diff
@@ -0,0 +1,180 @@
+) -> List[
+    Union[
+        DestinationV1_1_preview,
+        WebhookDestinationV1_1_preview,
+        AdxDestinationV1_1_preview,
+    ]
+]:
+    provider = CentralDestinationProvider(
+        cmd=cmd, app_id=app_id, api_version=api_version, token=token
+    )
+
+    return provider.list_dataExport_destinations(central_dns_suffix=central_dns_suffix)
+
+
+def add_dataExport_destination(
+    cmd,
+    app_id: str,
+    destination_id: str,
+    type,
+    display_name,
+    url=None,
+    cluster_url=None,
+    database=None,
+    table=None,
+    header_customizations=None,
+    authorization=None,
+    token=None,
+    central_dns_suffix=CENTRAL_ENDPOINT,
+    api_version=ApiVersion.v1_1_preview.value,
+) -> Union[
+    DestinationV1_1_preview, WebhookDestinationV1_1_preview, AdxDestinationV1_1_preview
+]:
+    destination = {
+        "id": destination_id,
+        "type": type,
+        "displayName": display_name,
+    }
+
+    if type == DestinationType.Webhook.value:
+        if not url:
+            raise CLIError(
+                "url parameter is required when creating webhook destination."
+            )
+        destination.update({"url": url})
+        if header_customizations is not None:
+            destination.update(
+                {
+                    "headerCustomizations": utility.process_json_arg(
+                        header_customizations, argument_name="header"
+                    )
+                }
+            )
+
+    if type == DestinationType.AzureDataExplorer.value:
+        if not cluster_url:
+            raise CLIError(
+                "cluster-url is required when creating azure data explorer destination."
+            )
+        if not database:
+            raise CLIError(
+                "database is required when creating azure data explorer destination."
+            )
+        if not table:
+            raise CLIError(
+                "table is required when creating azure data explorer destination."
+            )
+        destination.update(
+            {"clusterUrl": cluster_url, "database": database, "table": table}
+        )
+
+    if authorization:
+        destination.update(
+            {
+                "authorization": utility.process_json_arg(
+                    authorization, argument_name="authorization"
+                )
+            }
+        )
+    else:
+        if type != DestinationType.Webhook.value:
+            raise CLIError(
+                "authorization is required when creating non webhook destination."
+            )
+
+    provider = CentralDestinationProvider(
+        cmd=cmd, app_id=app_id, api_version=api_version, token=token
+    )
+
+    return provider.add_dataExport_destination(
+        destination_id=destination_id,
+        payload=destination,
+        central_dnx_suffix=central_dns_suffix,
+    )
+
+
+def update_dataExport_destination(
+    cmd,
+    app_id: str,
+    destination_id: str,
+    content: str,
```

Comment:
What kind of content change is allowed? For example, can a person change a destination from a webhook to an ADX cluster?
If a person is using an ADX cluster, can they change tables? This may need to be broken up

Reply:
Yes, type is not allowed to be changed, I had help summary to declare that on update command, for export update, the source is not allowed to be changed.

Reply:
Also would want update and add to support more or less the same functionality (unless certain updates are not allowed)